File: mypypkg/mypypkg/calculate.py
def add(a: float, b: float) -> float:
    """Return a result of a + b.

    Args:
        a (float)
        b (float)

    Returns:
        float: A result of a + b.
    """
    return a + b


def sub(a: float, b: float) -> float:
    """Return a result of a - b.

    Args:
        a (float)
        b (float)

    Returns:
        float: A result of a - b.
    """
    return a - b


def mul(a: float, b: float) -> float:
    """Return a result of a * b.

    Args:
       a (float)
       b (float)

    Returns:
       float: A result of a * b.
    """
    return a * b


def div(a: float, b: float) -> float:
    """Return a result of a / b.

    Args:
        a (float)
        b (float)

    Returns:
        float: A result of a / b.

    Raises:
        ValueError: b must not be 0.
    """
    if b == 0:
        raise ValueError('zero division is not allowed.')

    return a / b

File: mycpppkg/src/calculate.cpp
#include <cassert>

#include "mycpppkg/calculate.hpp"

namespace mycpppkg
{
  float add(const float a, const float b)
  {
    return a + b;
  }

  float sub(const float a, const float b)
  {
    return a - b;
  }

  float mul(const float a, const float b)
  {
    return a * b;
  }

  float div(const float a, const float b)
  {
    assert(b != 0 && "zero division is not allowed.");
    return a / b;
  }
} // namespace mycpppkg
